File: fibonacci/Python/fibo_cache.py
cache = {}

def fibo_cache(number):
    if number in cache:
        return cache[number]
    cache[number] = _fibo(number)
    return cache[number]

def _fibo(number):
    if number < 2:
        return 1
    return fibo_cache(number-1) + fibo_cache(number-2)
    

File: fibonacci/Go/monothread/fibo_cache.go
package main

var cache map[uint64]uint64 = make(map[uint64]uint64)

func fibo_cache(number uint64) uint64 {
	val, ok := cache[number]
	if ok {
		return val
	}
	cache[number] = _fibo(number)
	return cache[number]
}

func _fibo(number uint64) uint64 {
	if number < 2 {
		return number
	}
	return fibo_cache(number-1) + fibo_cache(number-2)
}
